src/main/java/org/jmxtrans/agent/properties/PropertiesLoaderImpl.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2010-2016 the original author or authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package org.jmxtrans.agent.properties;
+
+import org.jmxtrans.agent.util.Preconditions2;
+import org.jmxtrans.agent.util.io.IoRuntimeException;
+import org.jmxtrans.agent.util.io.Resource;
+import org.jmxtrans.agent.util.io.ResourceFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.annotation.Nonnull;
+
+/**
+ * {@link PropertiesLoader} based on a {@link Resource}.
+ *
+ * @author <a href="mailto:cleclerc@cloudbees.com">Cyrille Le Clerc</a>
+ * @author Kristoffer Erlandsson
+ */
+public class PropertiesLoaderImpl implements PropertiesLoader {
```

Comment:
I'm not fan of "*Impl" names as well but I was not sure of `ResourcePropertiesLoader`. I renamed as you have recommended.